pkg/crc/config/settings_test.go

```diff
@@ -0,0 +1,142 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/crc-org/crc/pkg/crc/constants"
+	"github.com/crc-org/crc/pkg/crc/logging"
+	"github.com/crc-org/crc/pkg/crc/preset"
+	"github.com/crc-org/crc/pkg/crc/version"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func newInMemoryConfig() (*Config, error) {
+	cfg := New(NewEmptyInMemoryStorage(), NewEmptyInMemorySecretStorage())
+
+	RegisterSettings(cfg)
+
+	return cfg, nil
+}
+
+func TestCPUValidate(t *testing.T) {
+	cfg, err := newInMemoryConfig()
+	require.NoError(t, err)
+
+	assert.Equal(t, SettingValue{
+		Value:     version.GetDefaultPreset().String(),
+		Invalid:   false,
+		IsDefault: true,
+		IsSecret:  false,
+	}, cfg.Get(Preset))
+
+	defaultCPUs := constants.GetDefaultCPUs(version.GetDefaultPreset())
+	_, err = cfg.Set(CPUs, defaultCPUs-1)
+	require.Error(t, err)
+	assert.Equal(t, SettingValue{
+		Value:     defaultCPUs,
+		Invalid:   false,
+		IsDefault: true,
+		IsSecret:  false,
+	}, cfg.Get(CPUs))
+
+	_, err = cfg.Set(Preset, preset.Podman)
+	require.NoError(t, err)
+	_, err = cfg.Set(CPUs, defaultCPUs-1)
+	require.NoError(t, err)
+	assert.Equal(t, SettingValue{
+		Value:     defaultCPUs - 1,
+		Invalid:   false,
+		IsDefault: false,
+		IsSecret:  false,
+	}, cfg.Get(CPUs))
+}
+
+func TestSetPreset(t *testing.T) {
+	cfg, err := newInMemoryConfig()
+	require.NoError(t, err)
+
+	_, err = cfg.Set(Preset, preset.Podman)
+	require.NoError(t, err)
+	_, err = cfg.Set(Memory, 10000)
```

Comment:
CI/tests are failing because of this, when we set memory, the memory of the system is checked, and the runners only have 8GB. Hard to workaround in the tests as the minimum memory amount for OpenShift is 9GB. I guess I'll remove this part of the test, it's not the key part.
This will have to wait until next week though.

```
=== RUN   TestSetPreset
    settings_test.go:63: 
        	Error Trace:	/root/rpmbuild/BUILD/crc-2.19.0-4.12.13/pkg/crc/config/settings_test.go:63
        	Error:      	Received unexpected error:
        	            	Value '10000' for configuration property 'memory' is invalid, reason: only 7.281GB of memory found (10.49GB required)
        	Test:       	TestSetPreset
--- FAIL: TestSetPreset (0.00s)
```

Reply:
Changing the cpu/memory to default we should use `podman` and `microshift` preset instead using `openshift` which fails because of high memory requirement.

Reply:
I agree. Public CI's should only concern with the smaller presets we offer.

Reply:
Using microshift will paper over this problem for now, but it will break in the same way the day we get a worker with less than 4G, or the day we raise the microshift requirements.

Reply:
I started changing the test to use the microshift preset instead of openshift, but this does not allow me to test `crc config unset preset`, which is one of the bugs I'm fixing. I'll try to handle this differently.